vunit/simulator_interface.py

```diff
@@ -214,13 +215,19 @@ def __compile_source_file(self, source_file, printer):
 
         return True
 
-    def compile_source_files(self, project, printer=NO_COLOR_PRINTER, continue_on_error=False):
+    def compile_source_files(self, project, printer=NO_COLOR_PRINTER, continue_on_error=False, target_files=None):
         """
         Use compile_source_file_command to compile all source_files
+        param: target_files: Given a list of SourceFiles only these and dependent files are compiled
         """
         dependency_graph = project.create_dependency_graph()
```

Comment:
If this was contained in a single call to Project class there would not need to be an extra test of the SimulatorInterface class